interfaces/repo.go

```diff
@@ -710,26 +704,6 @@ func (r *Repository) disconnect(plug *Plug, slot *Slot) {
 	if len(r.plugSlots[plug]) == 0 {
 		delete(r.plugSlots, plug)
 	}
-	for i, plugRef := range slot.Connections {
```

Comment:
Hmm? Don't you have to manage the connection state somehow?

Reply:
The state is maintained in the repo (slotPlugs, plugSlots maps); the only use of slot.Connections and plug.Connections was for direct marshaling in the api, they are now handled there.

Reply:
Ah, indeed, I didn't notice the type change above.